app/main/posts/views/filters/filters-dropdown.directive.js

```diff
@@ -39,8 +48,13 @@ function FiltersDropdown(PostFilters, ModalService, $rootScope) {
             // @TODO Prevent the user from creating one if they somehow manage to get to this point without being logged in
             $scope.savedSearch.user_id = $rootScope.currentUser ? $rootScope.currentUser.userId : null;
             ModalService.openTemplate('<saved-search-editor saved-search="savedSearch"></saved-search-editor>', 'set.create_savedsearch', 'star', $scope, false, false);
-
-            //ModalService.openTemplate('<savedsearch-create></savedsearch-create>', 'nav.saved_searches', '/img/iconic-sprite.svg#star', $scope.$parent, true, true);
+        };
+        $scope.editSavedSearchModal = function () {
+            $scope.savedSearch = $scope.filtersVar.saved_search;
+            $scope.savedSearch.filter = PostFilters.getActiveFilters($scope.filtersVar);
+            // @TODO Prevent the user from creating one if they somehow manage to get to this point without being logged in
```

Comment:
At this point, $scope.filtersVar.saved_search returns as an empty object in the $watch function, yet it still is populated correctly if you check the value within the editSavedSearchModal. If there is any asynchronous code, that could be causing the $watch method to mess up.